### PR TITLE
generate appmesh control code, implement appmesh output syncer

### PR DESCRIPTION
**Pull request by ilackarms**

### CODE REVIEW

File: pkg/mesh-discovery/translation/mesh/detector/mesh_detector.go [hunk 1]
```diff
@@ -1,30 +1,33 @@
...
 	"github.com/solo-io/service-mesh-hub/pkg/api/discovery.smh.solo.io/v1alpha2"
-	appsv1 "k8s.io/api/apps/v1"
 )
 
-// a deployment MeshDetector detects Mesh control plane instance (e.g. Pilot)
+// a deployment MeshDetector detects Mesh control plane instances (e.g. Pilot)
 // in a k8s Deployment.
 // If detection fails, an error is returned
 // If no mesh is detected, nil is returned
 // Separate Detectors are implemented for different Mesh types / versions.
```
Comment (harveyxia) [inline pkg/mesh-discovery/translation/mesh/detector/mesh_detector.go:13]: comment needs updating

File: pkg/mesh-discovery/translation/mesh/detector/mesh_detector.go [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
 type MeshDetector interface {
-	DetectMesh(deployment *appsv1.Deployment) (*v1alpha2.Mesh, error)
+	DetectMeshes(in input.Snapshot) (v1alpha2.MeshSlice, error)
 }
 
 // wrapper for multiple mesh detectors.
 // returns the first successfully detected mesh
```
Comment (harveyxia) [inline pkg/mesh-discovery/translation/mesh/detector/mesh_detector.go:19]:
```suggestion
// returns all detected meshes
```
```suggestion
// returns the first successfully detected mesh
```

File: pkg/mesh-networking/translation/appmesh/appmesh_networking_translator.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+		appmeshOutputs appmesh.Builder,
+		reporter reporting.Reporter,
+	)
+}
+
+type appmeshTranslator struct {
+	totalTranslates int // TODO(ilackarms): metric
+}
+
+func NewappmeshTranslator() Translator {
```
Comment (harveyxia) [inline pkg/mesh-networking/translation/appmesh/appmesh_networking_translator.go:30]:
```suggestion
func NewAppmeshTranslator() Translator {
```
```suggestion
func NewappmeshTranslator() Translator {
```

File: pkg/api/discovery.smh.solo.io/input/snapshot.go [hunk 4]
```diff
@@ -39,6 +47,9 @@ import (
 // the snapshot of input resources consumed by translation
 type Snapshot interface {
 
+	// return the set of input Meshes
+	Meshes() appmesh_k8s_aws_v1beta2_sets.MeshSet
```
Comment (harveyxia) [inline pkg/api/discovery.smh.solo.io/input/snapshot.go:51]: @ilackarms This name is a ambiguous, I wonder if we can either 1) extend skv2 to provide an optional name alias for the snapshot, or 2) group snapshot resources under a named GVK interface?

**Comment by joekelley** [comment]
ran into #968 a few times here, in one case for all istio runs at once

**Review by harveyxia — APPROVED** [review]

**Pull request merged**